File: Algorithmique appliquée/chapitre3/exercice4.cpp
/*  programme premier(exercice4)
*   but: controle si un nombre est premier utilisant une bibliotheque rudimentaire
*   date: 24/10/2016
*/
#include <iostream>
#include <maBiblio.h>

using namespace std;

/* fonction estPremier
*  but : indique si le nombre est premier
*  entrée : val (entier) reçoit le nombre entier à tester
*  sortie : bool premier indiquant si le test est positif ou negatif
*/

int main(int argc, char *argv[]){
    unsigned int val;

    cout << "Entrez un nombre entier(0 pour sortir) = ";
    cin >> val;
    while(val != 0){
        cout << boolalpha << estPremier(val) << "\n" << endl;
        cout << "Entrez un nombre entier(0 pour sortir) = ";
        cin >> val;
    }
    return 0;
}

File: Algorithmique appliquée/chapitre3/python/exercice4.py
# -*-coding:utf-8 -*
# Programme exercice 4
# reprise de l'exercice 6 de la séquence 1 avec une bibliotheque
# date : 24/10/2016

import os
import math
import sys
sys.path.append("..\\bibliotheque")
from maBiblio import *


# saisie du premier nombre à tester
val = int(input("entrez un nombre entier (0 pour finir) : "))

# boucle générale sur les tests
while val != 0 :
    # affichage du message correspondant au nombre
    if premier(val) :
        print(val, " est premier")
    else :
        print(val, " n'est pas premier")

    # saisie d'un nombre à tester
    val = int(input("entrez un nombre entier (0 pour finir) : "))


os.system("pause")
